front/lib/llm/providers/openai.ts

```diff
@@ -0,0 +1,163 @@
+import { OpenAI } from "openai";
+import type {
+  ResponseInputContent,
+  ResponseInputItem,
+} from "openai/resources/responses/responses.mjs";
+
+import { LLM } from "@app/lib/llm/llm";
+import type { LLMStreamEvent, ProviderMetadata } from "@app/lib/llm/types";
+import type {
+  Content,
+  RenderedModelConversationTypeMultiActions,
+} from "@app/types";
+
+export class OpenAILLM extends LLM {
+  private client: OpenAI;
+  private baseMetadata: ProviderMetadata<"openai"> = {
+    provider: "openai",
+    model: this.modelId,
+    metadata: {},
+  };
+
+  constructor({
+    temperature,
```

Comment:
I guess for this one we need sth like { temperature: number } | { reasoningEffort: 'low' | 'medium' | 'high' }